### ISSUE TITLE
Train & Save Clean CIFAR-10 ResNet-18 test

### ISSUE THREAD

**Issue by pelumitegbe** [oss-slu/mithridatium#1]
## Summary
Train a clean ResNet-18 on CIFAR-10 and save the best checkpoint to models/resnet18_clean.pth. This model will be used to extract embeddings and validate the Spectral Signatures defense which we will implement in Sprint 2

**Acceptance Criteria**

-  [ ] models/resnet18_clean.pth saved.

-  [ ] Test accuracy ≥70% (or documented if lower due to timebox).

-  [ ] Log file contains: seed, hyperparams, train/val metrics, final test acc.

-  [ ] Loader test command runs successfully.

## Tasks that need to be completed for this feature
 - Add training script to: scripts/train_resnet18.py.

 - Script arguments(for reproducability): --epochs, --batch-size, --lr, --seed, --out models/resnet18_clean.pth.

For example: A script could be:
python3 scripts/train_resnet18.py --epochs 30 --batch-size 128 --lr 0.1 --seed 42 --out models/resnet18_clean.pth

 - Implement train/val split (e.g., 45k/5k).Save best model by val acc.

 - Log metrics to stdout and save final test accuracy to logs/train_clean.txt.

 - Save best checkpoint (by val acc) to models/resnet18_clean.pth.

 - Verify loader compatibility:
    python -c "from mithridatium.loader import load_resnet18; \
           load_resnet18('models/resnet18_clean.pth'); print('ok')"

## Additional Context

- Target accuracy: ≥70% on CIFAR-10 test is acceptable for baseline (timebox epochs; note the result).
- Use standard augmentation (random crop/flip) and ImageNet normalization (to match our evaluator transforms).
Example of Random crop:
import torchvision.transforms as transforms

transform_train = transforms.Compose([
    transforms.RandomCrop(32, padding=4),   # crop randomly within 4 pixels
    transforms.RandomHorizontalFlip(),      # flip left/right
    transforms.ToTensor(),
    transforms.Normalize((0.485, 0.456, 0.406),
                         (0.229, 0.224, 0.225))  # ImageNet stats
])

- Set a fixed seed for reproducibility and print it in logs.

**Comment by Arsalanjdev** [comment]
Hi. Assign it to me.

**Comment by Arsalanjdev** [comment]
I've trained and saved a 84% accuracte model (with the size of 40mb) and its performance log. But the .gitignore entries prevent me from pushing the model and the log into git. should I remove those entries in .gitignore or is it the expected behavior?
